pkg/controller/mutators/stats_reporter_test.go

```diff
@@ -37,31 +37,31 @@ func TestReportMutatorIngestionRequest(t *testing.T) {
 	}
 
 	// Count test
-	row, err := checkData(mutatorIngestionCountMetricName, wantRowLength)
+	rows, err := checkData(mutatorIngestionCountMetricName, wantRowLength)
```

Comment:
These checks feel complex enough that it'd be better to just use cmp.Diff(). That gets us type differences, length, and numbers for free. It also provides a single logical place for defining the full expected object and provides output pretty-printed in a way we don't have to do ourselves.

Reply:
good call, will update

Reply:
Going to hold off on this change until after we resolve [this other thread](https://github.com/open-policy-agent/gatekeeper/pull/1714#discussion_r761536594), as it may involve deleting code I'd be updating for this comment.

Reply:
Sounds good!

Reply:
Will transition to `cmp.Diff`.  I think all the mutator tests would benefit from this approach.

Reply:
TLDR: I tried out some options here and it turns that the juice just isn't worth the squeeze.  I tried the obvious solution and it's not working.  Feel free to suggest an alternative if you know one.

Discussion:

So, it turns out that every metric row has a `time.time` in it.  That makes exact comparisons unavailable in the test.  

But, `cmp.Diff` has options for that.  See `IgnoreFields()` below:

```golang
	want := []*view.Row{
		{
			Data: &view.CountData{
				Value: wantIngestionCount,
			},
			Tags: []tag.Tag{
				{
					Key:   mutatorStatusKey,
					Value: string(MutatorStatusActive),
				},
			},
		},
	}

	if diff := cmp.Diff(want, got, cmpopts.IgnoreFields(view.CountData{}, "Start")); diff != "" {
		t.Errorf("data mismatch (-want +got):\n%s", diff)
	}
```

But, this doesn't work.  `Data` is actually an interface, which is my guess for why.

```
        -       s"{ { {status active} }&{0001-01-01 00:00:00 +0000 UTC 2} }",
        +       s"{ { {status active} }&{2021-12-06 16:53:07.706460136 -0800 PST m=+0.018122743 2} }",
```

With the obvious use of the feature not working, I don't think it's worth further investigation.  It _would_ make cleaner tests, but we could always try to solve this problem should we find ourselves needing to dramatically scale up the tests.  As we've completed the scoped metrics work for mutation, I think we're better off leaving this alone.

Reply:
SGTM